### PR TITLE
IP-242 - Add .snyk file.

### PR DESCRIPTION
**Pull request by owenlittlejohns**
## Description

This PR builds on some work that @flamingbear has done in other repositories to add a .snyk file that will ensure Snyk uses the correct version of Python when building an environment to scan the Harmony Regridding Service dependency tree. This will avoid false positives in our Snyk scans.

This change does not impact the service and only relates to our CI/CD (so it doesn't need a release or updated notes in the change log).

(Side note - we should separately bump the version of Python used from 3.9 to something more like 3.11, but that can be separate to this PR)

## Jira Issue ID

N/A

## Local Test Steps

* Check the version of Python listed in `docker/service.Dockerfile` matches the one in `.snyk`.
* After merging, check the version of Python Snyk uses is correct.

## PR Acceptance Checklist
* ~~Jira ticket acceptance criteria met.~~
* ~~`CHANGELOG.md` updated to include high level summary of PR changes.~~
* ~~`docker/service_version.txt` updated if publishing a release.~~
* [x] Tests added/updated and passing.
* [x] Documentation updated (if needed).

### CODE REVIEW

File: .snyk [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Snyk (https://snyk.io) policy file, patches or ignores known vulnerabilities.
+version: v1.25.0
+language-settings:
+  python: "3.9"
```
[the remark this answers is not in the document]
Comment (flamingbear) [reply .snyk:4]: was coming to ask why not bump the service version too. but this is fine

**Review by flamingbear — APPROVED** [review]
It'll be nice to bump the service version too, but this works.

File: .snyk (hunk 1 above)
[the remark this answers is not in the document]
Comment (owenlittlejohns) [reply .snyk:4]: I'll try and take a look at bumping the service in the next day or so. Now that I'm reading this thread back, I'm feeling like I was being lazy.

[the remark this answers is not in the document]
Comment (owenlittlejohns) [reply .snyk]: Alrighty, @flamingbear - it's all bumped to Python 3.11. Probably needs some more testing, likely running the regression test notebook against Harmony in a Box.

File: bin/extract-release-notes.sh [hunk 2]
```diff
@@ -2,22 +2,31 @@
 ###############################################################################
```
Comment (owenlittlejohns) [inline bin/extract-release-notes.sh:2]: I copied this over from HyBIG, because I also tweaked the `CHANGELOG.md` to adhere to Keep A Changelog.

**Comment by owenlittlejohns** [comment]
@flamingbear - nice catch on the documentation notebook. I've tweaked that now. I bumped the dependencies. It looks like it all runs nicely now, but I had to specify another dependency version (which was otherwise installed) to squash a _bunch_ of deprecation warnings.

**Pull request merged**